import_addressbase/import_csv.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+
+import csv
+from elasticsearch import Elasticsearch
+from elasticsearch.client import IndicesClient
+from elasticsearch.helpers import bulk
+from itertools import groupby
+from operator import itemgetter
+
+from import_addressbase.record_types import Header, BLPU, DPA
+
+
+INDEX_NAME = 'landregistry'
+
+HEADER_ID = 10  # Header record                (contains entry date)
+BLPU_ID = 21    # Basic Land and Property Unit (contains coordinates)
+DPA_ID = 28     # Delivery Point Address       (contains addresses)
+
+# the record identifier column index is common to all record types
+RECORD_IDENTIFIER = 0
+# these column indexes are common to the other record types we're interested in
+CHANGE_TYPE = 1
+CHANGE_TYPE_CODE = 2
+UPRN = 3
+
+ADDRESS_KEY_FIELDS = ['organisation_name', 'sub_building_name', 'building_name',
+                      'building_number', 'dependent_thoroughfare_name',
+                      'thoroughfare_name', 'double_dependent_locality',
+                      'dependent_locality', 'post_town', 'postcode']
+
+TYPE_TO_INDEX_MAPPING = [
+    ('property', 'addressKey'),
+    ('propertyByPostcode', 'postcode'),
+]
+
+
+def make_es_mappings(client):
+    no_index_properties = {
+        'uprn': {'type': 'string', 'index': 'no'},
+        'organisationName': {'type': 'string', 'index': 'no'},
+        'departmentName': {'type': 'string', 'index': 'no'},
+        'subBuildingName': {'type': 'string', 'index': 'no'},
+        'buildingName': {'type': 'string', 'index': 'no'},
+        'buildingNumber': {'type': 'string', 'index': 'no'},
+        'dependentThoroughfareName': {'type': 'string', 'index': 'no'},
+        'thoroughfareName': {'type': 'string', 'index': 'no'},
+        'doubleDependentLocality': {'type': 'string', 'index': 'no'},
+        'dependentLocality': {'type': 'string', 'index': 'no'},
+        'postTown': {'type': 'string', 'index': 'no'},
+        'postcode': {'type': 'string', 'index': 'no'},
+        'addressKey': {'type': 'string', 'index': 'no'},
+        'entryDatetime': {'type': 'date',
+                          'format': 'date_time_no_millis',
+                          'index': 'no'},
+    }
+
+    def make_es_mapping(doc_type, index_field):
+        mapping = {
+            doc_type: {'properties': no_index_properties}
+        }
+        mapping[doc_type]['properties'][index_field]['index'] = 'not_analyzed'
```

Comment:
For the purpose of readability, it may be better to set the index in `no_index_properties` first and then prepare `mapping`, so that we would have something like `no_index_properties[index_field]['index'] = 'not_analyzed'` and then `mapping` definition as it is. What do you think?